pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -435,3 +453,146 @@ func (c *EtcdProxyController) handleObject(obj interface{}) {
+		caConfigMapCopy.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+	}
+
+	if !equality.Semantic.DeepEqual(caConfigMap, caConfigMapCopy) {
+		_, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).Update(caConfigMapCopy)
+		if err != nil {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createEtcdProxyServerCertsSecret creates Secret in controller namespace with Etcd Proxy Server certificate and key.
+func (c EtcdProxyController) createEtcdProxyServerCertsSecret(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// Secret for the etcd proxy server certs in controller namespace.
+	certsSecret, err := c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+		Get(etcdProxyServerCertsSecret(etcdstorage), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		data := map[string]string{
+			"server.pem":     EtcdProxyServerCert,
+			"server-key.pem": EtcdProxyServerKey,
+		}
+		_, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+			Create(newSecret(etcdstorage, etcdProxyServerCertsSecret(etcdstorage), c.config.ControllerNamespace, data))
+		if err != nil {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+		return nil
+	}
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+
+	certsSecretCopy := certsSecret.DeepCopy()
+	certsSecretCopy.StringData = map[string]string{
+		"server.pem":     EtcdProxyServerCert,
+		"server-key.pem": EtcdProxyServerKey,
+	}
+
+	// Secret found, we're checking OwnerRef to make sure controller owns it.
+	if !metav1.IsControlledBy(certsSecret, etcdstorage) {
+		certsSecretCopy.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+	}
+
+	if !equality.Semantic.DeepEqual(certsSecret, certsSecretCopy) {
+		_, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).Update(certsSecretCopy)
+		if err != nil {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+// updateAPIServerServingCAConfigMap updates the ConfigMap in the aggregated API server namespace with the CA certificate.
+func (c *EtcdProxyController) updateAPIServerServingCAConfigMap(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// Check are ConfigMap name and namespace provided.
+	for _, configMap := range etcdstorage.Spec.CACertConfigMap {
+		caConfigMap, err := c.kubeclientset.CoreV1().ConfigMaps(configMap.Namespace).
+			Get(configMap.Name, metav1.GetOptions{})
+		if err != nil {
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+			return err
```

Comment:
track your errors, but continue to the next configmap so you complete as much work as possible.  Then return an aggregate error.